bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -119,13 +120,21 @@ public MavenWorkspaceRepository() {
 		};
 
 		identificationAndVersionExpressionFunction = (bsn, version) -> {
-			RequirementBuilder builder = new RequirementBuilder(BND_INFO);
-			builder.addFilter("name", bsn, version.toString(), null);
 			return combiner.or( //
 				combiner.identity(
-					CapReqBuilder.createSimpleRequirement(IdentityNamespace.IDENTITY_NAMESPACE, bsn, version.toString())
+					new RequirementBuilder(IdentityNamespace.IDENTITY_NAMESPACE) //
+						.addFilter(new FilterBuilder().and()
+							.eq(IdentityNamespace.IDENTITY_NAMESPACE, bsn)
+							.eq("version", version)
+							.end())
 						.buildSyntheticRequirement()), //
-				combiner.identity(builder.buildSyntheticRequirement()));
+				combiner.identity( //
+					new RequirementBuilder(BND_INFO) //
+						.addFilter(new FilterBuilder().and()
+							.eq(Constants.NAME_ATTRIBUTE, bsn)
+							.eq(IdentityNamespace.CAPABILITY_VERSION_ATTRIBUTE, version)
```

Comment:
```suggestion
							.eq("version", version)
```

This is the `bnd.info` namespace.